### PR TITLE
Feature: Add getPluginPreparations function

### PR DESCRIPTION
**Pull request by josemarinas**
## Description

Please include a summary of the change and be sure you follow the contributions rules we do provide [here](./CONTRIBUTIONS.md)

Task ID: [OS-792](https://aragonassociation.atlassian.net/browse/OS-792)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them when possible.
- [ ] I have updated the `CHANGELOG.md` file in the root folder of the package after the `[UPCOMING]` title and before the latest version.
- [ ] I have tested my code on the test network.

[OS-792]: https://aragonassociation.atlassian.net/browse/OS-792?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/client/src/internal/types.ts [hunk 1]
```diff
@@ -200,7 +200,41 @@ export type SubgraphPluginUpdatePreparation = {
...
+  who: string;
+  condition: string;
+  permissionId: string;
+};
+
+export enum SubgraphPluginPermissionOperation {
+  GRANT = "Grant",
+  REVOKE = "Revoke",
+  GRANT_WITH_CONDITION = "GrantWithCondition",
+}
```
Comment (Rekard0) [inline modules/client/src/internal/types.ts]: empty line at the end needed

**Review by Rekard0 — DISMISSED** [review]

File: modules/client/src/types.ts [hunk 2]
```diff
@@ -473,3 +474,37 @@ export type DaoUpdateDecodedParams = InitializeFromParams & {
...
 export type IsPluginUpdateValidParams = IsUpdateParamsBase;
+export type PluginPreparationQueryParams = Pagination & {
+  sortBy?: PluginPreparationSortBy;
+  type?: PluginPreparationType;
+  pluginAddress?: string;
+  pluginRepoAddress?: string;
+  daoAddressOrEns?: string;
+};
+export enum PluginPreparationType {
+  UPDATE = "Update",
```
Comment (Michael-A-Heuer) [inline modules/client/src/types.ts:485]: No Uninstallation yet?

File: modules/client/src/types.ts [hunk 3]
```diff
@@ -473,3 +474,37 @@ export type DaoUpdateDecodedParams = InitializeFromParams & {
...
+  INSTALLATION = "Installation",
```
Comment (heueristik) [inline modules/client/src/types.ts]:
```suggestion
  INSTALLATION = "Installation",
  UPDATE = "Update",
  UNINSTALLATION = "Uninstallation",
```

Why no Uninstallation yet? Also, chronological ordering seems to make more sense for the enumeration .

**Review by heueristik — COMMENTED** [review]
Minor suggestion

**Review by Rekard0 — APPROVED** [review]

**Pull request merged**